File: June-18/05-06-18/java/01-47s-String.java

import java.io.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        while(t-- > 0)
        {
            String str  = input.readLine();
            int n = str.length(), cnt = 0;
            long ans = 0;
            for(int i = 0; i < n; i++)
            {
                char ch = str.charAt(i);
                if(ch == '$' || ch == '&')
                {
                    ans = ans + (n-i) + ((n-i)*cnt);
                    cnt = 0;
                }
                else
                    cnt += 1;
            }
            System.out.println(ans);
            
        }
        
    }
}

File: June-18/05-06-18/python/01-47s-String.py
for _ in range(int(input())):
    data = input()
    n = len(data)
    cnt = 0
    ans = 0
    for i in range(0, n):
        if(data[i] == '$' or data[i] == '&'):
            ans = ans + (n-i) + ((n-i)*cnt)
            cnt = 0
        else:
            cnt += 1
    print(ans)
